### ISSUE TITLE
(3rd) Implement all of the functionality (client and server) for a creating an entry

### ISSUE THREAD

**Issue by h4w5** [h4w5/wdinstagram_collaboration_challenge#6]
Here is what needs to be done:

Implement the full-stack for creating an entry!

You must:

1. **Write out the [AJAX request][request] to create the entry.**
2. **Write the [route][route] to accept that request.**
3. **Write the [route handler][handler]** that:
   - fails and returns a `422` status code if the entry data is malformed or insufficient,
   - persists the data in the database if it correct,
   - correctly handles the case of an undefined date being passed, and
   - returns a copy of the created entry when all goes well.
4. **Accepts the response and [renders the "show" page][render] for the new entry** when it succeeds.

When you are complete, it should look like:

<img src="https://i.imgur.com/0e9Obez.png" width="100"> &nbsp; **=> SHOULD BECOME =>** &nbsp; <img src="https://i.imgur.com/49n539M.png" width="100">

---

Note, the [submit function already exists][submit]:

```js
$entryForm.on("submit", function(evt) {
  evt.preventDefault();

  var author    = $("#author").val(),
      photoUrl  = $("#photo-url").val(),
      dateTaken = $("#date-taken").val();

  console.log("submit triggered!", author, photoUrl, dateTaken);
  createEntry({author: author, photoUrl: photoUrl, dateTaken: dateTaken});
});
```

[request]: https://github.com/h4w5/wdinstagram_challenge/blob/master/public/js/main.js#L90-L92
[route]: https://github.com/h4w5/wdinstagram_challenge/blob/master/routes/api_routes.js
[handler]: https://github.com/h4w5/wdinstagram_challenge/blob/master/controllers/entries.js#L19
[render]: https://github.com/h4w5/wdinstagram_challenge/blob/master/public/js/main.js#L121-L134
[submit]: https://github.com/h4w5/wdinstagram_challenge/blob/master/public/js/main.js#L140-L149

**Comment by jgescobar** [comment]
Happy easter, 

AAU I can create an entry 

What I have done so far is : 
created the route to create in api_routes.js
```
router.post( '/entries',           entriesController.create);
```
in entries.js 
I uncommented the 
```
module.exports = {
   create: create
}; 
```
and also did the following 
<img src="https://i.imgur.com/FF7AW0F.png" height="450" width="450">

and did the function below in main.js : 
<img src="https://i.imgur.com/39bHJs0.png" height="400" width="400">

but i get this error : 
<img src="https://i.imgur.com/Nr3Zzol.png" height="650" width="450">

I'm not sure what I can do to fix this error. I believe the error is in entries.js and main.js.

**Comment by jgescobar** [comment]
AAU I would like to create images 

What Ive tried: 
I updated my function create in entries.js to look like the below : 
```
var create = function(req, res, next) {
    console.log("ServerSide New Instagram!!", req.body);

var newEntry = new Entry(req.body)

newEntry.save(function(err, savedEntry) {
     if (err || !savedEntry) {
     return res.status(422).json({message: "422 Unprocessable Entity"});
    } else {
      res.json(savedEntry);
  }
 });
}
```
This is my new error: 
![] (https://i.imgur.com/4O2IDx4.png) 

Is this what Im supposed to get when the function in entries.js its written correctly ?

**Comment by h4w5** [comment]
This all looks great!

However, you aren't sending any data! For your AJAX call, after the method and URL, send the data in the form! It's not easy, but you guys have examples of how to get the values out of inputs and package them with the call.

**Comment by h4w5** [comment]
And yeah, your server looks like it's structured the right way.

**Comment by h4w5** [comment]
Alright!

Closed by #13.

Nicely done. Good questions, etc. One problem, when you don't send a date, it should use today as the default date, but instead we get:

![Failed to have a working date.](https://i.imgur.com/e9Qg7Hr.png)

Given that, I'll give you 0.75 points for this. All in all, that's 2.5/3, which is passing. Nicely done!
